editor/src/messages/tool/tool_messages/fill_tool.rs

```diff
@@ -71,37 +76,111 @@ enum FillToolFsmState {
 	Filling,
 }
 
+#[derive(Clone, Debug, Default)]
+struct FillToolData {
```

Comment:
We need to set the tolerance in the tool's control bar. It should range from 0 (only the identically colored pixels) to 1 (all pixels regardless of color). We should also have a checkbox for contiguous.